flaky test in util/gctuner

## Bug Report

### 1. Minimal reproduce step (Required)

```
$go test -count=10 -run Test* github.com/pingcap/tidb/util/gctuner
--- FAIL: TestGlobalMemoryTuner (2.22s)
    memory_limit_tuner_test.go:103:
                Error Trace:    /x/tarball/gctuner/tidb-master/util/gctuner/memory_limit_tuner_test.go:103
                Error:          Not equal:
                                expected: 858993459
                                actual  : 1181116006
                Test:           TestGlobalMemoryTuner
    memory_limit_tuner_test.go:64:
                Error Trace:    /x/tarball/gctuner/tidb-master/util/gctuner/memory_limit_tuner_test.go:64
                                                        /x/tarball/gctuner/tidb-master/util/gctuner/panic.go:522
                                                        /x/tarball/gctuner/tidb-master/util/gctuner/testing.go:871
                                                        /x/tarball/gctuner/tidb-master/util/gctuner/memory_limit_tuner_test.go:103
                Error:          Should be false
                Test:           TestGlobalMemoryTuner
--- FAIL: TestTuner (0.29s)
    tuner_test.go:57:
                Error Trace:    /x/tarball/gctuner/tidb-master/util/gctuner/tuner_test.go:57
                Error:          "285" is not less than or equal to "250"
                Test:           TestTuner
--- FAIL: TestGlobalMemoryTuner (1.67s)
    memory_limit_tuner_test.go:92:
                Error Trace:    /x/tarball/gctuner/tidb-master/util/gctuner/memory_limit_tuner_test.go:92
                Error:          Not equal:
                                expected: 1181116006
                                actual  : 858993459
                Test:           TestGlobalMemoryTuner
    memory_limit_tuner_test.go:64:
                Error Trace:    /x/tarball/gctuner/tidb-master/util/gctuner/memory_limit_tuner_test.go:64
                                                        /x/tarball/gctuner/tidb-master/util/gctuner/panic.go:522
                                                        /x/tarball/gctuner/tidb-master/util/gctuner/testing.go:871
                                                        /x/tarball/gctuner/tidb-master/util/gctuner/memory_limit_tuner_test.go:92
                Error:          Should be false
                Test:           TestGlobalMemoryTuner
--- FAIL: TestGlobalMemoryTuner (1.66s)
    memory_limit_tuner_test.go:92:
                Error Trace:    /x/tarball/gctuner/tidb-master/util/gctuner/memory_limit_tuner_test.go:92
                Error:          Not equal:
                                expected: 1181116006
                                actual  : 858993459
                Test:           TestGlobalMemoryTuner
    memory_limit_tuner_test.go:64:
                Error Trace:    /x/tarball/gctuner/tidb-master/util/gctuner/memory_limit_tuner_test.go:64
                                                        /x/tarball/gctuner/tidb-master/util/gctuner/panic.go:522
                                                        /x/tarball/gctuner/tidb-master/util/gctuner/testing.go:871
                                                        /x/tarball/gctuner/tidb-master/util/gctuner/memory_limit_tuner_test.go:92
                Error:          Should be false
                Test:           TestGlobalMemoryTuner
--- FAIL: TestFinalizer (0.01s)
    finalizer_test.go:40:
                Error Trace:    /x/tarball/gctuner/tidb-master/util/gctuner/finalizer_test.go:40
                Error:          Not equal:
                                expected: 9
                                actual  : 8
                Test:           TestFinalizer
```
<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/util/gctuner/tuner_test.go
+++ b/util/gctuner/tuner_test.go
@@ -39,7 +39,7 @@ func TestTuner(t *testing.T) {
 	for i := 0; i < 100; i++ {
 		runtime.GC()
 		require.Eventually(t, func() bool { return maxGCPercent.Load() == tn.getGCPercent() },
-			500*time.Microsecond, 50*time.Microsecond)
+			1*time.Second, 50*time.Microsecond)
 	}
 
 	// 1/4 threshold
@@ -56,9 +56,9 @@ func TestTuner(t *testing.T) {
 	for i := 0; i < 100; i++ {
 		runtime.GC()
 		require.Eventually(t, func() bool { return tn.getGCPercent() >= minGCPercent.Load() },
-			500*time.Microsecond, 50*time.Microsecond)
+			1*time.Second, 50*time.Microsecond)
 		require.Eventually(t, func() bool { return tn.getGCPercent() <= maxGCPercent.Load()/2 },
-			500*time.Microsecond, 50*time.Microsecond)
+			1*time.Second, 50*time.Microsecond)
 	}
 
 	// 3/4 threshold
@@ -67,7 +67,7 @@ func TestTuner(t *testing.T) {
 	for i := 0; i < 100; i++ {
 		runtime.GC()
 		require.Eventually(t, func() bool { return minGCPercent.Load() == tn.getGCPercent() },
-			500*time.Microsecond, 50*time.Microsecond)
+			1*time.Second, 50*time.Microsecond)
 	}
 
 	// out of threshold
@@ -76,7 +76,7 @@ func TestTuner(t *testing.T) {
 	for i := 0; i < 100; i++ {
 		runtime.GC()
 		require.Eventually(t, func() bool { return minGCPercent.Load() == tn.getGCPercent() },
-			500*time.Microsecond, 50*time.Microsecond)
+			1*time.Second, 50*time.Microsecond)
 	}
 }
 

--- a/util/gctuner/tuner_test.go
+++ b/util/gctuner/tuner_test.go
@@ -17,6 +17,7 @@ package gctuner
 import (
 	"runtime"
 	"testing"
+	"time"
 
 	"github.com/stretchr/testify/require"
 )
@@ -37,7 +38,8 @@ func TestTuner(t *testing.T) {
 	runtime.GC()
 	for i := 0; i < 100; i++ {
 		runtime.GC()
-		require.Equal(t, maxGCPercent.Load(), tn.getGCPercent())
+		require.Eventually(t, func() bool { return maxGCPercent.Load() == tn.getGCPercent() },
+			500*time.Microsecond, 50*time.Microsecond)
 	}
 
 	// 1/4 threshold
@@ -53,8 +55,10 @@ func TestTuner(t *testing.T) {
 	runtime.GC()
 	for i := 0; i < 100; i++ {
 		runtime.GC()
-		require.GreaterOrEqual(t, tn.getGCPercent(), minGCPercent.Load())
-		require.LessOrEqual(t, tn.getGCPercent(), maxGCPercent.Load()/2)
+		require.Eventually(t, func() bool { return tn.getGCPercent() >= minGCPercent.Load() },
+			500*time.Microsecond, 50*time.Microsecond)
+		require.Eventually(t, func() bool { return tn.getGCPercent() <= maxGCPercent.Load()/2 },
+			500*time.Microsecond, 50*time.Microsecond)
 	}
 
 	// 3/4 threshold
@@ -62,7 +66,8 @@ func TestTuner(t *testing.T) {
 	runtime.GC()
 	for i := 0; i < 100; i++ {
 		runtime.GC()
-		require.Equal(t, minGCPercent.Load(), tn.getGCPercent())
+		require.Eventually(t, func() bool { return minGCPercent.Load() == tn.getGCPercent() },
+			500*time.Microsecond, 50*time.Microsecond)
 	}
 
 	// out of threshold
@@ -70,7 +75,8 @@ func TestTuner(t *testing.T) {
 	runtime.GC()
 	for i := 0; i < 100; i++ {
 		runtime.GC()
-		require.Equal(t, minGCPercent.Load(), tn.getGCPercent())
+		require.Eventually(t, func() bool { return minGCPercent.Load() == tn.getGCPercent() },
+			500*time.Microsecond, 50*time.Microsecond)
 	}
 }
 

